### PR TITLE
Validate gaslimit on pendingstate

### PR DESCRIPTION
**Pull request by donequis**

### CODE REVIEW

File: rskj-core/src/main/java/org/ethereum/core/Block.java [hunk 1]
```diff
@@ -701,6 +701,10 @@ public BigInteger getMinGasPriceAsInteger() {
         return (this.getMinimumGasPrice() == null) ? null : new BigInteger(1, this.getMinimumGasPrice());
     }
 
+    public BigInteger getGasLimitAsInteger() {
```
Comment (tinchou) [inline rskj-core/src/main/java/org/ethereum/core/Block.java]: Please use `BigIntegers.fromUnsignedByteArray`

File: rskj-core/src/main/java/co/rsk/net/handler/txvalidator/TxValidatorGasLimitValidator.java [hunk 2]
```diff
@@ -32,6 +33,6 @@
     @Override
     public boolean validate(Transaction tx, AccountState state, BigInteger gasLimit, BigInteger minimumGasPrice, long bestBlockNumber) {
         BigInteger txGasLimit = tx.getGasLimitAsInteger();
-        return gasLimit.compareTo(txGasLimit) >= 0;
+        return  txGasLimit.compareTo(gasLimit) <= 0;
```
Comment (tinchou) [inline rskj-core/src/main/java/co/rsk/net/handler/txvalidator/TxValidatorGasLimitValidator.java]: Please use `BIUtil.isMoreThan(gasLimit, txGasLimit)` or `!BIUtil.isLessThan(txGasLimit, gasLimit)`

Comment (aeidelman) [reply rskj-core/src/main/java/co/rsk/net/handler/txvalidator/TxValidatorGasLimitValidator.java]: Semantic of those methods is not very clear IMHO

File: rskj-core/src/main/java/org/ethereum/core/Block.java (hunk 1 above)
Comment (aeidelman) [reply rskj-core/src/main/java/org/ethereum/core/Block.java]: Done

File: rskj-core/src/main/java/co/rsk/net/handler/txvalidator/TxValidatorGasLimitValidator.java (hunk 2 above)
Comment (tinchou) [reply rskj-core/src/main/java/co/rsk/net/handler/txvalidator/TxValidatorGasLimitValidator.java]: Would be great to have extension methods

**Pull request merged**